src/plots/cartesian/axis_defaults.js

```diff
@@ -64,6 +65,10 @@ module.exports = function handleAxisDefaults(containerIn, containerOut, coerce,
         }
     }
 
+    containerOut._initialCategories = axType === 'category' ?
+        orderedCategories(letter, containerIn.categorymode, containerIn.categorylist, options.data) :
```

Comment:
@etpinard tl;dr blocking question for you at the end.

As an aid to thinking about goals, here's a summary of what coercions I'm planning:

1. If `categorymode` is not one of those in the attributes definition, it'll be coerced to the default `trace` value.
2. No matter what `categorymode` is specified or coerced in item#1, if a `categorylist` is provided and it is an array, `categorymode` will be coerced into `array`.

... well it looks like the two things you listed :-) My original take was that `categorymode`, as a switch, drives the show in an unambiguous way, and the user can flip-flop among modes without worrying about removing and readding (or commenting/uncommenting) the `categorylist`  property. Even now I'm vague about the case where `categorymode` is `category ascending` and `categorylist` is also given. Moreover, it renders the `categorymode === 'array'` redundant (documentational at best) since the presence of `categorylist` takes precedence. Maybe we should even do away with `categorymode === 'array'`, making `categorymode` and `categorylist` mutually exclusive?

Taking it one step further, shouldn't we just have *one* property, e.g. called `categoryorder` that can be `category ascending`, `category descending` or an array?

I'm now plunging into the basic coercion so we don't have an unexpected enumerative value but your input to the above question would be useful. It's not a big deal to unify the two properties and at least technically, it would be less ambiguous.